### PR TITLE
feat: Remove special handling of promoted tags in errors replacer

### PR DESCRIPTION
**Pull request by lynnagara**
This removes all special handling of promoted tags in the errors replacer
and instead uses the values from tags.key and tags.value like our
unpromoted tags.

Unlike in the events storage, most of our promoted tags are low cardinality
columns now. This is causing a bug in ClickHouse -
https://github.com/ClickHouse/ClickHouse/issues/16171.

Our workaround for now is to not use any of our promoted columns for
errors replacements; event replacements are unchanged.

This fixes SNUBA-22N in Sentry.

### CODE REVIEW

File: snuba/datasets/storages/errors.py [hunk 1]
```diff
@@ -164,8 +164,8 @@
     replacer_processor=ErrorsReplacer(
         schema=schema,
         required_columns=required_columns,
-        tag_column_map={"tags": promoted_tag_columns, "contexts": {}},
-        promoted_tags={"tags": list(promoted_tag_columns.keys()), "contexts": []},
+        tag_column_map={"tags": {}, "contexts": {}},
+        promoted_tags={"tags": [], "contexts": []},
```
Comment (fpacifici) [inline snuba/datasets/storages/errors.py:168]: Am I missing something or, when deleting a promoted tag, now we would not be setting the column to NULL ?

https://github.com/getsentry/snuba/blob/master/snuba/datasets/errors_replacer.py#L415-L416

the `promoted_tags` field is used to decide which columns to set to NULL when the tag is removed.

Comment (lynnagara) [reply snuba/datasets/storages/errors.py:168]: No, you are right - I forgot about this. Thanks.

File: snuba/datasets/errors_replacer.py [hunk 2]
```diff
@@ -379,6 +383,7 @@ def process_delete_tag(
     all_columns: Sequence[FlattenedColumn],
     tag_column_map: Mapping[str, Mapping[str, str]],
     promoted_tags: Mapping[str, Sequence[str]],
+    state_name: Optional[ReplacerState],
```
Comment (fpacifici) [inline snuba/datasets/errors_replacer.py:386]: This is not guaranteed to accurately tell apart errors from events. It is more of a coincidence now. It is a flag used to identify the variable to rely on to save the groups in redis.
I would add another flag to this class properly named. Something like `disable_prewhere`. So these two can change independently as they are not semantically tied. There is no design reason for prewhere to be disabled when we use the ERROR state in redis.

**Review by fpacifici — COMMENTED** [review]
I would not use ReplacerState to decide when to disable the PREWHERE. I would add a new flag instead. Otherwise this is fine.

**Review by fpacifici — APPROVED** [review]

**Pull request merged**